sql/src/main/java/io/crate/operation/merge/RamAccountingPageIterator.java

```diff
@@ -0,0 +1,116 @@
+package io.crate.operation.merge;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.crate.breaker.RamAccountingContext;
+import io.crate.breaker.RowAccounting;
+import io.crate.data.Row;
+import io.crate.operation.projectors.sorting.OrderingByPosition;
+import io.crate.planner.PositionalOrderBy;
+import io.crate.planner.node.dql.MergePhase;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+
+import java.util.function.Consumer;
+
+/**
+ * Wraps a PagingIterator and uses {@link io.crate.breaker.RamAccountingContext} to apply the circuit breaking logic
+ */
+public class RamAccountingPageIterator implements PagingIterator<Integer, Row> {
+
+    @VisibleForTesting
+    final PagingIterator<Integer, Row> delegatePagingIterator;
+    private final RowAccounting rowAccounting;
+
+    private RamAccountingPageIterator(PagingIterator<Integer, Row> delegatePagingIterator,
+                                      RowAccounting rowAccounting) {
+        this.delegatePagingIterator = delegatePagingIterator;
+        this.rowAccounting = rowAccounting;
+    }
+
+    public static PagingIterator<Integer, Row> create(MergePhase mergePhase,
```

Comment:
Would move this back into PagingIterators or maybe even into the interface itself. Having it here is missleading as the RamAccountingPageIterator isn't always used